tests/tests_fabric/utilities/test_throughput.py

```diff
@@ -0,0 +1,260 @@
+from unittest import mock
+from unittest.mock import Mock, call
+
+import pytest
+import torch
+from lightning.fabric import Fabric
+from lightning.fabric.plugins import Precision
+from lightning.fabric.utilities.throughput import (
+    Throughput,
+    ThroughputMonitor,
+    _MonotonicWindow,
+    get_available_flops,
+    measure_flops,
+)
+
+from tests_fabric.helpers.runif import RunIf
+from tests_fabric.test_fabric import BoringModel
+
+
+@RunIf(min_torch="2.1")
+def test_measure_flops():
+    with torch.device("meta"):
+        model = BoringModel()
+        x = torch.randn(2, 32)
+    model_fwd = lambda: model(x)
+    model_loss = lambda y: y.sum()
+
+    fwd_flops = measure_flops(model, model_fwd)
+    assert isinstance(fwd_flops, int)
+
+    fwd_and_bwd_flops = measure_flops(model, model_fwd, model_loss)
+    assert isinstance(fwd_and_bwd_flops, int)
+    assert fwd_flops < fwd_and_bwd_flops
+
+
+def test_available_flops(xla_available):
+    with mock.patch("torch.cuda.get_device_name", return_value="NVIDIA H100 PCIe"):
+        flops = get_available_flops(torch.device("cuda"), torch.bfloat16)
+    assert flops == 1.513e15 / 2
+
+    with pytest.warns(match="not found for 'CocoNut"), mock.patch("torch.cuda.get_device_name", return_value="CocoNut"):
+        assert get_available_flops(torch.device("cuda"), torch.bfloat16) is None
+
+    with pytest.warns(match="t4' does not support torch.bfloat"), mock.patch(
+        "torch.cuda.get_device_name", return_value="t4"
+    ):
+        assert get_available_flops(torch.device("cuda"), torch.bfloat16) is None
+
+    from torch_xla.experimental import tpu
+
+    assert isinstance(tpu, Mock)
+    tpu.get_tpu_env.return_value = {"TYPE": "V4"}
+
+    flops = get_available_flops(torch.device("xla"), torch.bfloat16)
+    assert flops == 275e12
+
+    tpu.get_tpu_env.return_value = {"TYPE": "V1"}
+    with pytest.warns(match="not found for TPU 'V1'"):
+        assert get_available_flops(torch.device("xla"), torch.bfloat16) is None
+
+    tpu.reset_mock()
+
+
+def test_throughput():
+    # required args only
+    throughput = Throughput()
+    throughput.update(time=2.0, samples=2)
+    assert throughput.compute() == {"time": 2.0, "samples": 2}
+
+    # different lengths and samples
+    with pytest.raises(RuntimeError, match="same number of samples"):
+        throughput.update(time=2.1, samples=3, lengths=4)
+
+    # lengths and samples
+    throughput = Throughput(window_size=2).update(time=2, samples=2, lengths=4).update(time=2.5, samples=4, lengths=8)
```

Comment:
Why is it important to support chaining? The doc says it follows the interface of torchmetrics.Metric, but Metric.update() returns None, not Self.

Reply:
It's not important, I simply found it convenient. Do you want me to revert it?

Reply:
I don't see a use case for it, so I would remove it. Calling .update().update().update() is not meaningful in my opinion.